backend/collaboration-service/app/src/server.ts

```diff
@@ -0,0 +1,60 @@
+import http from "http"
+import app from "./app"
+import { Server } from "socket.io"
+import { ChangeSet } from "@codemirror/state";
+import { getUpdateInfo, updateDocInfo, getDocumentInfo } from "./models/rooms.model";
+import { handleGetDocument, handlePullUpdates, handlePushUpdates, handleResetDocument } from "./socket/doc";
+
+const port = process.env.port || 8005;
+
+export const server = http.createServer(app,);
+
+const io = new Server(server, {
+    // options
+    cors: {
+        origin: "*",
+        methods: ["GET", "POST"]
+    }
+})
+
+io.on('connection', (socket) => {
+
+    const roomId = socket.handshake.query.roomId;
+
+    if (typeof roomId !== "string") {
+        socket.emit('error', 'Invalid roomId provided')
```

Comment:
can abstract the messages/events into constants

Reply:
Resolved [9c17ce9](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/pull/18/commits/9c17ce960f24723b076597f4006805bed0a49565)